File: tutorials/data_persistence/c++11/temperature_publisher.cxx
//
// (c) 2024 Copyright, Real-Time Innovations, Inc.  All rights reserved.
//
//  RTI grants Licensee a license to use, modify, compile, and create derivative
//  works of the Software.  Licensee has the right to distribute object form
//  only for use with RTI products.  The Software is provided "as is", with no
//  warranty of any type, including any warranty for fitness for any purpose.
//  RTI is under no obligation to maintain or support the Software.  RTI shall
//  not be liable for any incidental or consequential damages arising out of the
//  use or inability to use the software.
//

#include <iostream>
#include <thread>
#include <rti/rti.hpp>
#include "temperature.hpp"

void publish_temperature(const std::string& sensor_name)
{
    dds::domain::DomainParticipant participant(0);
    dds::topic::Topic<Temperature> topic(participant, "Temperature");
    dds::pub::DataWriter<Temperature> writer(topic);

    Temperature temp_reading;
    temp_reading.sensor_name(sensor_name);
    for (int i = 0; i < 1000; i++) {
        temp_reading.degrees((rand() % 10) + 30);
        writer.write(temp_reading);
        std::this_thread::sleep_for(std::chrono::seconds(1));
    }
}

int main(int argc, char **argv)
{
    std::string sensor_name = (argc > 1) ? argv[1] : "Sensor1";
    publish_temperature(sensor_name);
}

File: tutorials/data_persistence/py/temperature_publisher.py
#
# (c) 2024 Copyright, Real-Time Innovations, Inc.  All rights reserved.
#
# RTI grants Licensee a license to use, modify, compile, and create derivative
# works of the Software solely for use with RTI products.  The Software is
# provided "as is", with no warranty of any type, including any warranty for
# fitness for any purpose. RTI is under no obligation to maintain or support
# the Software.  RTI shall not be liable for any incidental or consequential
# damages arising out of the use or inability to use the software.
#

import sys
import random
from time import sleep

import rti.connextdds as dds
from temperature import Temperature

def publish_temperature(sensor_name: str):
    participant = dds.DomainParticipant(domain_id=0)
    topic = dds.Topic(participant, "Temperature", Temperature)
    writer = dds.DataWriter(topic)

    temp_reading = Temperature(sensor_name)
    for _ in range(1000):
        temp_reading.degrees = random.uniform(30, 40)
        writer.write(temp_reading)
        sleep(1)


if __name__ == "__main__":
    sensor_name = sys.argv[1] if len(sys.argv) > 1 else "Sensor1"
    try:
        publish_temperature(sensor_name)
    except KeyboardInterrupt:
        pass
